continuous-deployment/readme.md

```diff
@@ -1,7 +1,142 @@
+
+#### Modeling your Release Pipeline
+
+It's critical to model your test and release process to establish a common understanding between the application engineers and customer stakeholders. Specifically aligning expectations for how many cloud environments need to be pre-provisioned as well as defining sign-off gate roles and responsibilities.
+
+![image](https://user-images.githubusercontent.com/7635865/89290276-426ca200-d61e-11ea-9c79-bd183dc72050.png)
+
+##### The release pipeline model should depict:
+
+* All stages an application change would have to go through before it is released to production.
+* All release gate controls.
+* Customer-specific Cloud RBAC groups which have the authority to approve release candidates per environment.
+
+#### Release Pipeline Stages
+
+The stages within your release workflow are ultimately testing a version of your application to validate it can be released in accordance to your acceptance criteria. The release pipeline should account for the following conditions:
+
+* Release Selection: The developer carrying out application testing should have the capability to select which release version to deploy to the testing environment.
+* Environment Readiness - Clean the target cloud environment so it is ready to host a new version of the application.
+* Deployment - Release the application binary(s) to the target cloud environment.
+* Configuration - Applications should be configured in a consistent manner across all your environments. This configuration is applied at the time of deployment. Application Secrets should not be exposed within the runtime environment. Sensitive data like app secrets and certificates should be mastered in Key Vault and sourced internally within the application itself. We encourage 12 Factor principles, especially when it comes to [configuration management](https://12factor.net/config).
+* Data Migration - Pre populate application state and/or data records which is needed for your runtime environment. This may also include test data required for your end-to-end integration test suite.
+* Deployment smoke test. Your smoke test should also verify that your application is pointing to the correct configuration (e.g. production pointing to a UAT Database).
+* Perform any manual or automated acceptance test scenarios.
+* Approve the release gate to promote the application version to the target cloud environment. This promotion should also include the environment's configuration state (e.g. new env settings, feature flags, etc).
+
+#### Pre-production releases
+
+Application release candidates should be deployed to a staging environment similar to production for carrying out final manual/automated tests (_including capacity testing_). Your production and staging / pre-prod cloud environments should be setup at the beginning of your project.
+
+Application warm up should be a quantified measurement that's validated as part of your pre-prod smoke tests.
```

Comment:
Adding more details on what is considered part of application warm up would be helpful here.